internal/cmd/cmd_type.go

```diff
@@ -0,0 +1,56 @@
+package cmd
+
+import (
+	"github.com/dicedb/dice/internal/object"
+	dstore "github.com/dicedb/dice/internal/store"
+	"github.com/dicedb/dicedb-go/wire"
+)
+
+var cType = &DiceDBCommand{
+	Name:      "TYPE",
+	HelpShort: "Determine data type of the value stored at a specified key",
+	Eval:      evalTYPE,
+}
+
+func init() {
+	commandRegistry.AddCommand(cType)
+}
+
+func evalTYPE(c *Cmd, s *dstore.Store) (*CmdRes, error) {
+	if len(c.C.Args) != 1 {
+		return cmdResNil, errWrongArgumentCount("TYPE")
+	}
+
+	key := c.C.Args[0]
+	obj := s.Get(key)
+
+	if obj == nil {
+		return &CmdRes{R: &wire.Response{
+			Value: &wire.Response_VStr{
+				VStr: "none",
+			},
+		}}, nil
+	}
+
+	var typeStr string
+	switch oType := obj.Type; oType {
+	case object.ObjTypeString, object.ObjTypeInt, object.ObjTypeByteArray:
+		typeStr = "string"
+	case object.ObjTypeDequeue:
+		typeStr = "list"
+	case object.ObjTypeSet:
+		typeStr = "set"
+	case object.ObjTypeHashMap:
+		typeStr = "hash"
+	case object.ObjTypeSortedSet:
+		typeStr = "zset"
+	default:
+		typeStr = "non-supported type"
+	}
```

Comment:
Why not just define a String method? I recommend add the following method in `object.go` file and call the `String` function here.

```go
// String returns the name of the object type as a string
func (ot ObjectType) String() string {
	names := [...]string{
		"string",
		"",
		"",
		"json",
		"bytes",
		"int",
		"set",
		"hashmap",
		"sortedset",
		"countminsketch",
		"bf",
		"dequeue",
		"hll",
		"float",
	}
	if ot < ObjectType(len(names)) {
		return names[ot]
	}
	return "Unknown"
}
```